backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/DBPerformanceMonitoringHelper.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.compiere.util;
+
+import de.metas.monitoring.adapter.NoopPerformanceMonitoringService;
+import de.metas.monitoring.adapter.PerformanceMonitoringService;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.service.ISysConfigBL;
+import org.compiere.SpringContextHolder;
+
+import java.util.concurrent.Callable;
+
+public class DBPerformanceMonitoringHelper
```

Comment:
we already have org.adempiere.ad.dao.impl.QueryStatisticsLogger which is measuring performances....
what we would have to do is to improve it and make it work with micrometer.

The big advantage of QueryStatisticsLogger is that
* it's quite performant
* already works
* it's converting all executed SQLs..... which this code does not, e.g. not sure if ResultSet.executeQuery() is converted...

Reply:
DB.executeUpdate() includes some ResultSet.executeQuery(), but in general it isn't monitored. After QuerryStatisticLogger is added DB.executeUpdate monitoring i think can be removed. 
DB.prepareStatement() isn't included in QuerryStatisticLogger and we could keep it and the basic Framework for more DB.java monitoring, if needed later

Reply:
disagree... let's talk